src/frame.js

```diff
@@ -0,0 +1,30 @@
+var Frame = function() {
+  this.firstBowl = null;
+  this.secondBowl = 0;
+  this.result;
+  this.spare = false;
+  this.strike = false;
+};
+ 
+Frame.prototype.receiveBowl = function(pinsKnockedOver) {
+  if (this.firstBowl === null && pinsKnockedOver === 10) {
+    this.firstBowl = pinsKnockedOver;
+    this.secondBowl = 0;
+    this.calculateResult();
+  } else if (this.firstBowl != null) {
+    this.secondBowl += pinsKnockedOver;
+    this.calculateResult();
+  } else {
+    this.firstBowl = pinsKnockedOver;
+  };
+};
+
+Frame.prototype.calculateResult = function() {
```

Comment:
`result` is not set when there is a strike or spare.  Is this correct?  This could lead to unexpected behaviour and a disappointed consumer